server/routes/allocate-key-workers/controller.ts

```diff
@@ -0,0 +1,49 @@
+import { Request, Response } from 'express'
+import KeyworkerApiService from '../../services/keyworkerApi/keyworkerApiService'
+import { lastNameCommaFirstName } from '../../utils/formatUtils'
+import LocationsInsidePrisonApiService from '../../services/locationsInsidePrisonApi/locationsInsidePrisonApiService'
+
+export class AllocateKeyWorkerController {
+  constructor(
+    private readonly keyworkerApiService: KeyworkerApiService,
+    private readonly locationsApiService: LocationsInsidePrisonApiService,
+  ) {}
+
+  GET = async (req: Request, res: Response): Promise<void> => {
+    const prisonCode = res.locals.user.activeCaseLoad!.caseLoadId!
+    const records = await this.keyworkerApiService.searchPrisoners(req, prisonCode, {
+      query: req.query['query']?.toString() || '',
+      cellLocationPrefix: req.query['location']?.toString() || '',
+    })
+
+    const keyworkers = await this.keyworkerApiService.getKeyworkerMembers(req, prisonCode, { status: 'ACTIVE' })
+    const locations = await this.locationsApiService.getResidentialLocations(req, prisonCode)
+
+    const withoutKeyworker = req.query['withoutKeyworker'] === 'true'
+    res.render('allocate-key-workers/view', {
+      query: req.query['query'],
+      location: req.query['location'],
+      withoutKeyworker,
+      records: records.filter(o => withoutKeyworker || o.keyworker),
```

Comment:
Just spotted you're doing this filtering in the UI. You can use the `excludeActiveAllocations` request parameter for the API call. That's what it was designed for. If it's true it will not return anyone with an active keyworker allocation so I think you can pass `withoutKeyworker` as the value.